vector-im/element-android
# [Location sharing] Fix crash when starting/stopping a live when offline (PSF-1124)

## Type of change


## Content


Catching errors which may be thrown during start/stop process of a live location share. The errors are forwarded to the UI: an error dialog is shown in both `TimelineFragment` and `LocationLiveMapViewFragment`.

## Motivation and context


Closes #6315 
Fixes https://github.com/matrix-org/element-android-rageshakes/issues/39071v

## Screenshots / GIFs





<img src="https://user-images.githubusercontent.com/46314705/173844121-5fe43052-f58f-4daa-b1a2-93403509f1ad.png" width=300/>

## Tests



- Enable airplane mode
- Start a live location share
- Check an error dialog is shown in the timeline screen
- Disable airplane mode
- Start a live location share
- Check the live is correctly started
- Enable airplane mode
- Stop the live
- Check an error dialog is shown in the timeline screen
- Go to the map screen
- Stop the live
- Check an error dialog is shown in the map screen
- Disable airplane mode
- Stop the live
- Check the live is correctly stopped

## Tested devices

- OS version(s): Android 11

## Checklist
